### PR TITLE
Moving to pyproject.toml

### PR DESCRIPTION
**Pull request by Seluj78**
As you might know, `setup.py install` is deprecated and legacy, and when installing your project, we get this error:

```
  DEPRECATION: json-log-formatter is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
```

This PR updated the install method.

### CODE REVIEW

**Comment by marselester** [comment]
Thank you, that's awesome!

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+[project]
+name = "JSON-log-formatter"
+description = 'JSON log formatter'
+authors = [
+    {name = "Marsel Mavletkulov", email = "marselester@ya.ru"},
+]
+readme = "README.rst"
+version = '0.5.2'
```
Comment (marselester) [inline pyproject.toml]: Let's keep `0.5.1` version as in the `setup.py`.

**Comment by Seluj78** [comment]
You're welcome :D I've been using your project for the past year or so for work, and I thought contributing that was the least I could do !

File: pyproject.toml (hunk 1 above)
Comment (Seluj78) [reply pyproject.toml]:
```suggestion
version = '0.5.1'
```

File: setup.py [hunk 2]
```diff
@@ -1,21 +0,0 @@
-from setuptools import setup
```
Comment (marselester) [inline setup.py]: Please don't delete this file as it can be useful in legacy setups.

Comment (Seluj78) [reply setup.py]: Done !

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+requires-python = ">=3.6"
```
Comment (marselester) [inline pyproject.toml]: Let's remove this requirement if it's possible (this library works with Python 2 as well).

**Comment by Seluj78** [comment]
I might also suggest adding the `__version__` tag in your `__init__.py` file, and loading the version in `setup.py` and `pyproject.toml` from there dynamically, which will allow you to change the version only in the `__init__.py` file

File: pyproject.toml (hunk 3 above)
Comment (Seluj78) [reply pyproject.toml]: Done

**Comment by marselester** [comment]
It's ok, I will update the version manually to keep things simple.

**Comment by marselester** [comment]
Thanks again for contributing 👍.

**Pull request merged**